File: final_dataset_A/generated_code/chatgpt/java/Submit_customer_feedback_020_Secure.java
import java.util.ArrayList;
import java.util.List;
import java.util.regex.Pattern;

public class Submit_customer_feedback_020_Secure {

    static class Feedback {
        String name;
        String email;
        int rating;
        String message;

        Feedback(String n, String e, int r, String m) {
            name = n;
            email = e;
            rating = r;
            message = m;
        }
    }

    static List<Feedback> feedbacks = new ArrayList<>();

    static final Pattern EMAIL =
            Pattern.compile("^[A-Za-z0-9+_.-]+@[A-Za-z0-9.-]+$");

    public static String submitFeedback(
            String name,
            String email,
            int rating,
            String message) {

        if (name == null || email == null || message == null)
            return "Required fields missing.";

        name = name.trim();
        email = email.trim();
        message = message.trim();

        if (name.isEmpty() || email.isEmpty() || message.isEmpty())
            return "Required fields missing.";

        if (!EMAIL.matcher(email).matches())
            return "Invalid email address.";

        if (rating < 1 || rating > 5)
            return "Rating must be between 1 and 5.";

        feedbacks.add(new Feedback(
                name,
                email,
                rating,
                message));

        return "Feedback submitted successfully.";
    }
}

File: final_dataset_A/generated_code/chatgpt/python/Submit_customer_feedback_020_Secure.py
import re


feedbacks = []


EMAIL_PATTERN = r"^[\w\.-]+@[\w\.-]+\.\w+$"


def submit_customer_feedback(
        name,
        email,
        rating,
        message):

    """
    Securely submits customer feedback.
    """

    try:

        if name is None or email is None or message is None:
            return "Required fields are missing."

        name = name.strip()
        email = email.strip()
        message = message.strip()

        if not name or not email or not message:
            return "Required fields are missing."

        if not re.match(EMAIL_PATTERN, email):
            return "Invalid email address."

        if not isinstance(rating, int):
            return "Invalid rating."

        if rating < 1 or rating > 5:
            return "Rating must be between 1 and 5."

        feedbacks.append({
            "name": name,
            "email": email,
            "rating": rating,
            "message": message
        })

        return "Feedback submitted successfully."

    except Exception:
        return "Unable to submit feedback."


if __name__ == "__main__":
    print(submit_customer_feedback(
        "Ahmed",
        "ahmed@example.com",
        5,
        "Excellent service!"
    ))